sys/compat/linprocfs/linprocfs.c

```diff
@@ -2092,6 +2092,140 @@ linprocfs_domax_map_cnt(PFS_FILL_ARGS)
 	return (0);
 }
 
+/*
+ * Filler function for proc/sysvipc/msg
+ */
+static int
+linprocfs_dosysvipc_msg(PFS_FILL_ARGS)
+{
+	struct msqid_kernel *msqids;
+	u_long id, msgmni;
+	size_t sz;
+	int error;
+
+	sbuf_printf(sb,
+	    "%10s %10s %4s  %10s %10s %5s %5s %5s %5s %5s %5s %10s %10s %10s\n",
+	    "key", "msqid", "perms", "cbytes", "qnum", "lspid", "lrpid",
+	    "uid", "gid", "cuid", "cgid", "stime", "rtime", "ctime");
+
+	error = kern_get_msqids(curthread, &msqids, &sz);
+	if (error != 0)
+		return (error);
+	msgmni = sz / sizeof(struct msqid_kernel);
```

Comment:
sz already contains the count of elements, not the size. No need to delete by sizeof().  Same for all other places.

Reply:
Fixed.

Reply:
Now it works!  It shows the same information for all SYSVIPC mechanisms!